internal/rbd/controllerserver.go

```diff
@@ -119,17 +120,24 @@ func (cs *ControllerServer) parseVolCreateRequest(
 		return nil, status.Error(codes.InvalidArgument, "empty imageFeatures parameter")
 	}
 
+	// get the owner of the PVC which is required for few encryption related operations
+	namespace := req.GetParameters()["csi.storage.k8s.io/pvc/namespace"]
```

Comment:
>It is cleaner to set the rbdVol.Owner as part of genVolFromVolumeOptions()

This is not possible because of in node plugin we are extracting the owner details from omap after calling  `genVolFromVolumeOptions` that is why i moved the  setting owner and calling `rv.initKMS` outside of the `genVolFromVolumeOptions` function

other changes are done except above